nupic/research/connections.py

```diff
@@ -238,6 +238,54 @@ def numSynapses(self):
     return len(self._synapses)
 
 
+  def write(self, proto):
+    protoCells = proto.init('cells', self.numCells)
+
+    i = 0
+    for cell in xrange(self.numCells):
+      segments = self.segmentsForCell(cell)
+      protoSegments = protoCells[cell].init('segments', len(segments))
+
+      j = 0
+      for segment in segments:
+        synapses = self.synapsesForSegment(segment)
+        protoSynapses = protoSegments[j].init('synapses', len(synapses))
+
+        k = 0
+        for synapse in synapses:
+          synapseData = self.dataForSynapse(synapse)
+          protoSynapse = protoSynapses[k]
+
+          protoSynapse.presynapticCell = synapseData.presynapticCell
+          protoSynapse.permanence = synapseData.permanence
+
+          k += 1
+
+        j += 1
+
+      i += 1
+
+
+  def read(self, proto):
```

Comment:
Switch to classmethod and create instance with `__new__` like this:
https://github.com/numenta/nupic/blob/6c78a70ccbba9e642864b2d9d81714f19b1cb7e1/nupic/encoders/category.py#L239